### ISSUE TITLE
Update empty state for workspace feeds

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#53091]
As part of an on-going update throughout the app, we are updating the empty state of workspace feeds to include the call-to-action directly within the empty state modal, like so. 

![image](https://github.com/user-attachments/assets/2c7ca7b9-f307-4bf6-af9e-d98d90f77717)

To summarize this empty state/action framework, it works like this:
 - If there is an empty state and we want you to take an action, we put a (green) button in the empty state
 - Once you take the action, we move the (green) button to where it'd normally live. 

![CleanShot 2024-11-25 at 23 12 44](https://github.com/user-attachments/assets/c07c425d-668d-49d4-861d-c8f8dacd992e)

cc @joekaufmanexpensify in case you want to assign yourself. I'm going to keep the internal label on this issue since it's since in development and hard to test.

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Update empty state for workspace feeds

### What is the root cause of that problem?

New feature

### What changes do you think we should make in order to solve the problem?


Add the `Assign card` button to `WorkspaceCompanyCardsFeedAddedEmptyPage` using `buttons` prop where `handleAssignCard` is taken from [here](https://github.com/Expensify/App/blob/2570632ce5be0391db55a9b1c9f55a074361564f/src/pages/workspace/companyCards/WorkspaceCompanyCardsListHeaderButtons.tsx#L53).

We might need to add `icon` prop to `EmptyStateComponent`'s `Button` and implement it [here](https://github.com/Expensify/App/blob/3822f77cf0b5fc8838f635711547dde55fece413/src/components/EmptyStateComponent/index.tsx#L108).

```
buttons={[{
    success: true,
    buttonText: translate('workspace.companyCards.assignCard'),
    buttonAction: handleAssignCard,
}]}
```

**Comment by joekaufmanexpensify** [comment]
@mkzie2 This is part of an ongoing project and is not currently soliciting volunteers (and doesn't have the `Help wanted` label). Please reference the contributing guidelines if you have any Q's!

**Comment by joekaufmanexpensify** [comment]
>cc @joekaufmanexpensify in case you want to assign yourself. I'm going to keep the internal label on this issue since it's since in development and hard to test.

Awesome. I will work on getting this assigned out!

**Comment by DylanDylann** [comment]
I can help as c+

**Comment by DylanDylann** [comment]
@JmillsExpensify @dukenv0307 Do we also apply this design for direct feed?

cc @mountiny

**Comment by mountiny** [comment]
@DylanDylann @dukenv0307 yes, majority of these changes should work same for both direct and custom feeds

**Comment by DylanDylann** [comment]
@jliexpensify No need payment here

**Comment by jliexpensify** [comment]
Oh ok - thanks for the heads up! Does that also apply for @dukenv0307?

**Comment by joekaufmanexpensify** [comment]
Left a comment on [the PR](https://github.com/Expensify/App/pull/53118#issuecomment-2512977968), I don't think the current solution did exactly what we were intending.

**Comment by mountiny** [comment]
Discussing [here](https://expensify.slack.com/archives/C06ML6X0W9L/p1733200020036699), seems like C+ cannot reproduce your issue

**Comment by mountiny** [comment]
@dukenv0307 @DylanDylann coming from here https://expensify.slack.com/archives/C06ML6X0W9L/p1733253221396639?thread_ts=1733200020.036699&cid=C06ML6X0W9L seems like @joekaufmanexpensify is still able to reproduce using direct feed on mobile. Can you please have a look?

**Comment by jliexpensify** [comment]
Checking here - does @dukenv0307 require payment?

**Comment by jliexpensify** [comment]
Also bumping @DylanDylann for the checklist

**Comment by DylanDylann** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production
  - [x] 2b. Reported on staging (deploy blocker)
  - [ ] 2c. Reported on both staging and production
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [x] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: No because this seems like a case wasn't considered in the document

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.


- [x] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal


### Test 1



1. Go to a new WS
2. Open the company card page
3. Add new Feed successful
4. The empty state view displays with a green Assign Card button in the empty state
5. Click on Assign Card button in the empty state and complete the assign card flow
6. The green Assign Card button move to the top right of page


### Test 2



1. Go to a new WS
2. Open the company card page
3. Add new Feed
4. And the feed is pending
5. The App shows a greyed assign card button at the top right of the page


Do we agree 👍 or 👎

**Comment by jliexpensify** [comment]
For Step 6, should it be:

> The green Assign Card button shows at the top right of page

because its not really moving right?

**Comment by jliexpensify** [comment]
^ @DylanDylann sorry forgot to tag you

**Comment by DylanDylann** [comment]
@jliexpensify Updated

**Comment by jliexpensify** [comment]
Thanks everyone, closing this!
